ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/AccountSettings.hs

```diff
@@ -0,0 +1,79 @@
+module Ariadne.UI.Qt.Widgets.Dialogs.AccountSettings
+  ( RenameHandler
+  , DeleteHandler
+  , runAccountSettings
+  ) where
+
+import qualified Data.Text as T
+import Graphics.UI.Qtah.Core.Types (QtCursorShape(..))
+import Graphics.UI.Qtah.Signal (connect_)
+
+import qualified Graphics.UI.Qtah.Gui.QCursor as QCursor
+import qualified Graphics.UI.Qtah.Widgets.QAbstractButton as QAbstractButton
+import qualified Graphics.UI.Qtah.Widgets.QBoxLayout as QBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QDialog as QDialog
+import qualified Graphics.UI.Qtah.Widgets.QHBoxLayout as QHBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QLabel as QLabel
+import qualified Graphics.UI.Qtah.Widgets.QLineEdit as QLineEdit
+import qualified Graphics.UI.Qtah.Widgets.QPushButton as QPushButton
+import qualified Graphics.UI.Qtah.Widgets.QWidget as QWidget
+
+import Ariadne.UI.Qt.UI
+import Ariadne.UI.Qt.Widgets.Dialogs.Util
+
+data AccountSettings =
+  AccountSettings
+    { accountSettings :: QDialog.QDialog
+    }
+
+type RenameHandler = Text -> IO ()
+type DeleteHandler = IO ()
+
+initAccountSettings :: Text -> RenameHandler -> DeleteHandler -> IO AccountSettings
+initAccountSettings currentName renameHandler deleteHandler = do
+  accountSettings <- QDialog.new
+  layout <- createLayout accountSettings
+
+  let headerString = toString $ T.toUpper "ACCOUNT SETTINGS"
+
+  QWidget.setWindowTitle accountSettings headerString
+
+  header <- QLabel.newWithText headerString
+  addHeader layout header
+
+  accountNameLabel <- QLabel.newWithText ("ACCOUNT NAME" :: String)
+  accountNameEdit <- QLineEdit.newWithText $ toString currentName
+
+  addRow layout accountNameLabel accountNameEdit
+
+  pointingCursor <- QCursor.newWithCursorShape PointingHandCursor
+  buttonsLayout <- QHBoxLayout.new
+  deleteButton <- QPushButton.newWithText ("Delete account" :: String)
+  QWidget.setCursor deleteButton pointingCursor
+
+  QBoxLayout.addStretch buttonsLayout
+  QBoxLayout.addWidget buttonsLayout deleteButton
+  QBoxLayout.addStretch buttonsLayout
+
+  QBoxLayout.addLayout layout buttonsLayout
+
+  QPushButton.setDefault deleteButton False
+  QPushButton.setAutoDefault deleteButton False
+
+  setProperty deleteButton ("dialogButtonRole" :: Text) ("textDangerButton" :: Text)
+
+  let
+    asettings = AccountSettings{..}
+
+  connect_ deleteButton QAbstractButton.clickedSignal $ \_ ->
+    deleteHandler >> QDialog.accept accountSettings
+  connect_ accountNameEdit QLineEdit.editingFinishedSignal $
```

Comment:
When I alt-tab account settings with something changed in `QLineEdit`, name of the account is renamed. Maybe it's better to add button to perform renaming or rename on _Enter_ click.

Reply:
I suppose renaming should be done only when `Enter` is pressed.